### ISSUE TITLE
[HL2DM] [Linux] Native Half-Life 2: Deathmatch does not run 

### ISSUE THREAD

**Issue by DenAkimov** [ValveSoftware/Source-1-Games#6556]
Native Half-Life 2: Deathmatch fails to run, outputting:
"FATAL: It appears hl2 was not launched within the Steam for Linux sniper runtime environment.
FATAL: Please consult documentation to ensure correct configuration, aborting."

**Comment by nixRidge** [comment]
Additionally, if I modify the hl2.sh launch script for Half-Life 2: Deathmatch to stop checking for the runtime environment, it complains about missing a launch binary.

In fact, I cannot find any binaries named hl2, hl2_linux, or hl2_linux64 in the Half-Life 2: Deathmatch directories. Anywhere.

**Comment by someone12421** [comment]
having the same issue, using arch linux

<details><summary>logs:</summary>

> [2024-11-17 22:03:43] chdir "/drive2/Games/SteamLibrary/steamapps/common/Half-Life 2 Deathmatch"
> [2024-11-17 22:03:43] ERROR: ld.so: object '/home/user/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
> [2024-11-17 22:03:43] Game Recording - would start recording game 320, but recording for this game is disabled
> [2024-11-17 22:03:43] Adding process 85880 for gameID 320
> [2024-11-17 22:03:43] ERROR: ld.so: object '/home/user/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
> [2024-11-17 22:03:43] ERROR: ld.so: object '/home/user/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
> [2024-11-17 22:03:43] WARNING: discarding _NET_WM_PID 6 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
> [2024-11-17 22:03:43] ERROR: ld.so: object '/home/user/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
> [2024-11-17 22:03:43] gamemodeauto: 
> [2024-11-17 22:03:43] ERROR: ld.so: object '/home/user/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
> [2024-11-17 22:03:43] gamemodeauto: 
> [2024-11-17 22:03:43] ERROR: ld.so: object '/home/user/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
> [2024-11-17 22:03:43] ERROR: ld.so: object '/home/user/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
> [2024-11-17 22:03:43] Adding process 85892 for gameID 320
> [2024-11-17 22:03:43] Adding process 85893 for gameID 320
> [2024-11-17 22:03:44] i386-linux-gnu-capsule-capture-libs: warning: Dependencies of /home/user/.local/share/Steam/ubuntu12_32/libVkLayer_steam_fossilize.so not found, ignoring: Missing dependencies: Could not find "librt.so.1" in LD_LIBRARY_PATH "/usr/lib:/drive2/Games/SteamLibrary/steamapps/common/Half-Life 2 Deathmatch", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
> [2024-11-17 22:03:44] i386-linux-gnu-capsule-capture-libs: warning: Dependencies of /home/user/.local/share/Steam/ubuntu12_32/steamoverlayvulkanlayer.so not found, ignoring: Missing dependencies: Could not find "libdl.so.2" in LD_LIBRARY_PATH "/usr/lib:/drive2/Games/SteamLibrary/steamapps/common/Half-Life 2 Deathmatch", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
> [2024-11-17 22:03:44] i386-linux-gnu-capsule-capture-libs: warning: Dependencies of /usr/lib32/dri/crocus_dri.so not found, ignoring: Missing dependencies: Could not find "libgbm.so.1" in LD_LIBRARY_PATH "/usr/lib:/drive2/Games/SteamLibrary/steamapps/common/Half-Life 2 Deathmatch", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
> [2024-11-17 22:03:44] i386-linux-gnu-capsule-capture-libs: warning: Dependencies of /usr/lib32/dri/nouveau_drv_video.so not found, ignoring: Missing dependencies: Could not find "libglapi.so.0" in LD_LIBRARY_PATH "/usr/lib:/drive2/Games/SteamLibrary/steamapps/common/Half-Life 2 Deathmatch", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
> [2024-11-17 22:03:44] pressure-vessel-wrap[85892]: W: Using glibc from provider system for some but not all architectures! Arbitrarily using provider locales.
> [2024-11-17 22:03:44] pressure-vessel-wrap[85892]: W: Using libdrm from provider system for some but not all architectures! Will take /usr/share/libdrm from provider.
> [2024-11-17 22:03:44] pressure-vessel-wrap[85892]: W: Using libGLX_mesa.so.0 from provider system for some but not all architectures! Will take /usr/share/drirc.d from provider.
> [2024-11-17 22:03:44] ERROR: ld.so: object 'libgamemodeauto.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
> [2024-11-17 22:03:44] ERROR: ld.so: object 'libgamemodeauto.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
> [2024-11-17 22:03:44] pid 85977 != 85973, skipping destruction (fork without exec?)
> [2024-11-17 22:03:44] pid 85982 != 85973, skipping destruction (fork without exec?)
> [2024-11-17 22:03:44] Adding process 85973 for gameID 320
> [2024-11-17 22:03:44] ERROR: ld.so: object 'libgamemodeauto.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
> [2024-11-17 22:03:44] ERROR: ld.so: object 'libgamemodeauto.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
> [2024-11-17 22:03:44] ERROR: ld.so: object 'libgamemodeauto.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
> [2024-11-17 22:03:44] pid 86004 != 85973, skipping destruction (fork without exec?)
> [2024-11-17 22:03:44] ERROR: ld.so: object 'libgamemodeauto.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
> [2024-11-17 22:03:44] 
> [2024-11-17 22:03:44] FATAL: It appears hl2 was not launched within the Steam for Linux sniper runtime environment.
> [2024-11-17 22:03:44] FATAL: Please consult documentation to ensure correct configuration, aborting.
> [2024-11-17 22:03:44] 
> [2024-11-17 22:03:44] Adding process 86003 for gameID 320
> [2024-11-17 22:03:44] Adding process 86005 for gameID 320
> [2024-11-17 22:03:44] Game Recording - game stopped [gameid=320]
> [2024-11-17 22:03:44] Removing process 86005 for gameID 320
> [2024-11-17 22:03:44] Removing process 86003 for gameID 320
> [2024-11-17 22:03:44] Removing process 85973 for gameID 320
> [2024-11-17 22:03:44] Removing process 85893 for gameID 320
> [2024-11-17 22:03:44] Removing process 85892 for gameID 320
> [2024-11-17 22:03:44] Removing process 85880 for gameID 320
> 

</details>

**Comment by someone12421** [comment]
seems like the prerelease, previous_build, and steam_legacy beta branches work. the "stable" version and the beta branch called "beta" dont work

**Comment by nixRidge** [comment]
Only _prerelease, previous_build, and steam_legacy_ will work on Linux & MacOS right now. Public & beta are Windows only for now.

[According to SteamDB](https://steamdb.info/app/320/depots/), the three legacy repos are all on the same build of the game. So for now, use any of these to play the game and pray the server you like to play on hasn't updated. Public & beta removed the launcher binaries for both Linux & MacOS.

![bilde](https://github.com/user-attachments/assets/450d7ca3-eaac-40bd-acf0-5862e88c6fa7)

**Comment by someone12421** [comment]
> Only _prerelease, previous_build, and steam_legacy_ will work on Linux & MacOS right now. Public & beta are Windows only for now.
> 
> [According to SteamDB](https://steamdb.info/app/320/depots/), the three legacy repos are all on the same build of the game. So for now, use any of these to play the game and pray the server you like to play on hasn't updated. Public & beta removed the launcher binaries for both Linux & MacOS.
> 
> ![bilde](https://private-user-images.githubusercontent.com/40672277/387002218-450d7ca3-eaac-40bd-acf0-5862e88c6fa7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE4NzcxMzEsIm5iZiI6MTczMTg3NjgzMSwicGF0aCI6Ii80MDY3MjI3Ny8zODcwMDIyMTgtNDUwZDdjYTMtZWFhYy00MGJkLWFjZjAtNTg2MmU4OGM2ZmE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE3VDIwNTM1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmMDQ2ZDZmYzc3OWE1ZmI4YWUxMWVmMWUzYjc5NzY3ZTdlNGRjNjYyYmFlYWQxODdmOTkzY2U3ZGQ1YWNlOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.acX31yVAyIjXpBOBtKCQYsj8blaXzWSCtfMkXiINBoM)

valve please fix (genuinely though, how tf did this happen?)

**Comment by utku-helvaci** [comment]
if you try to get the `hl2_linux` and `hl2.sh` it won't work due to missing `libtcmalloc_minimal.so.4` but also due to new build shipping with 64 bit binaries. old executable wants 32 bit:
```
/mnt/7e35b636-96c0-4070-b04f-ad2f69b6a203/SteamLibrary/steamapps/common/Half-Life 2 Deathmatch/hl2_linux: error while loading shared libraries: libtcmalloc_minimal.so.4: cannot open shared object file: No such file or directory
```
after copying `libtcmalloc_minimal.so.4` from portal or cs source:
```
AppFramework : Unable to load module sourcevr.so!
Using shader api: shaderapivk
 failed to dlopen /mnt/7e35b636-96c0-4070-b04f-ad2f69b6a203/SteamLibrary/steamapps/common/Half-Life 2 Deathmatch/bin/shaderapivk.so error=libdxvk_d3d9.so: wrong ELF class: ELFCLASS64
 failed to dlopen shaderapivk.so error=shaderapivk.so: wrong ELF class: ELFCLASS64
Failed to load DLL 'shaderapivk'
MaterialSystem cannot load: No ShaderAPIFactory!
System (VMaterialSystem081) failed during stage CONNECTION
```

**Comment by DiegoArmando-27** [comment]
> wrong ELF class: ELFCLASS64

It says that the `shaderapivk` library is 64-bit and 32-bit executable basically can't load it.

So yes, the updated HL2:DM version needs a new-built 64-bit executable.

**Comment by dhcpme** [comment]
On Fedora 41, _steam_legacy_ is the **only**  version that works for me.

**Comment by niobium93** [comment]
Copying the singleplayer HL2 `hl2_linux` binary and then launching it like this actually runs, but then immediately crashes.
`LD_LIBRARY_PATH="/home/andrius/.local/share/Steam/steamapps/common/Half-Life 2 Deathmatch/bin":$LD_LIBRARY_PATH ./hl2_linux -game hl2mp -steam`
[Log](https://gist.github.com/niobium93/818cea39939d008cf4315088d8461e85)

TL;DR: 
```
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_S_construct null not valid
Segmentation fault (core dumped)
```
To be clear the hl2mp client.so and server.so are still 32-bit, so you need a 32-bit launcher binary and 32-bit engine libraries. The ones in `/bin` are 32-bit, but for some reason there are also redundant 64-bit libraries in `/bin/linux64`. The `hl2.sh` shell script is made for launching the 64-bit engine (it tries to run `hl2_linux64` and prepends `/bin/linux64` to `LD_LIBRARY_PATH` instead of just `/bin`) so it obviously won't work unless the game code is actually rebuilt for 64-bit. Even if you put everything where it needs to be, the engine just crashes. It looks like no one even tried to run this on Linux or macOS, let alone play tested it...

**Comment by nixRidge** [comment]
> steam_legacy is the only version that works for me.

It's the only version that will work for anyone on Mac and Linux right now, they're both missing binaries.

If I'm to make an optimistic guess, maybe they did this intentionally to prevent people from launching the game, perhaps because they need more time to work on a fix for the other two clients.

Either way, I don't think throwing more logs at this is going to help either way. The binaries are missing, so that's it.

Now I'm no Valve developer, so I have no idea what their schedule is like other than they usually deploy updates and releases on Thursdays, so remember this probably won't be fixed until at minimum later this week **IF** they're currently working on a fix.
It'll probably take even longer if they even bother to fix it at all.

HL2DM doesn't make them any money so they could just as well ignore this and leave it Windows exclusive. Ironically.

**Comment by niobium93** [comment]
@nixRidge this was updated for the big Half-Life 2 20th Anniversary update. It definitely hasn't been broken on purpose, they could've just kept linux/macOS on the old depots before the update if it wasn't done.

**Comment by nixRidge** [comment]
In that case then yeah they goofed hard lol

**Comment by Lassebq** [comment]
#6528

**Comment by dhcpme** [comment]
The servers I normally play have upgraded to the 20th Anniversary Update so I can no longer play using _steam_legacy_. I'm hopeful this gets resolved soon since it's a much different feel when I boot/play under Windows and I live/work natively under Linux. I really do appreciate everyone's efforts to support us in a tiny, tiny minority.

**Comment by PQCraft** [comment]
hl2dm also doesn't work for me anymore.
The 'hl2.sh' script fails the `[ "$VERSION_CODENAME" != "sniper" ]` check. Disabling this check by commenting out the `exit 1` will start the game but it will die with an error saying "Could not load library client. Try restarting. If that doesn't work, verify the cache."

**Comment by wiretab** [comment]
Also experiencing this on Mint 21.3.

**Comment by Wolfyxon** [comment]
Same here
```
chdir "/run/media/wolfyxon/Acer/SteamLibrary/steamapps/common/Half-Life 2 Deathmatch"
ERROR: ld.so: object '/home/wolfyxon/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/wolfyxon/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/wolfyxon/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/wolfyxon/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
Game Recording - would start recording game 320, but recording for this game is disabled
Adding process 14268 for gameID 320
ERROR: ld.so: object '/home/wolfyxon/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/wolfyxon/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Adding process 14269 for gameID 320
Adding process 14270 for gameID 320
pid 14357 != 14351, skipping destruction (fork without exec?)
pid 14362 != 14351, skipping destruction (fork without exec?)
Adding process 14351 for gameID 320
pid 14384 != 14351, skipping destruction (fork without exec?)

FATAL: It appears hl2 was not launched within the Steam for Linux sniper runtime environment.
FATAL: Please consult documentation to ensure correct configuration, aborting.

Adding process 14383 for gameID 320
Adding process 14385 for gameID 320
Game Recording - game stopped [gameid=320]
Removing process 14385 for gameID 320
Removing process 14383 for gameID 320
Removing process 14351 for gameID 320
Removing process 14270 for gameID 320
Removing process 14269 for gameID 320
Removing process 14268 for gameID 320
```

**Comment by scorpius2k1** [comment]
+1 for Arch Linux, and what seems like on ALL Linux distributions, HL2DM is completely broken and fails to launch after the anniversary update. Half Life 2 single player works completely fine, but there appears to be no binary distribution for the Linux client at all for HL2 multiplayer, even though maps and several other files were updated. Only setting the beta option for the game to 'steam_legacy' client works, but is not a viable option any longer as many servers are unconnectable now with them being updated to the main branch.

Presumably, this also affects the Steam Deck as well with it being Linux based.

@kisak-valve as this critical issue has persisted for a week now, any updates or ETA on a solution? Thank you in advance.

**Comment by gegoxaren** [comment]
It is broken on Flatpak too.

**Comment by nixRidge** [comment]
I don't think any amount of "same here", logs, or worse, pinging Kisak (who is not a Valve developer) is going to help the situation other than bloating up the comments section. The binaries are gone, there's nothing else to do other than wait for them to come back.

**Comment by scorpius2k1** [comment]
> I don't think any amount of "same here", logs, or worse, pinging Kisak (who is not a Valve developer) is going to help the situation other than bloating up the comments section. The binaries are gone, there's nothing else to do other than wait for them to come back.

I don't think being silent and not sharing feedback on the issue will help either. At any rate, HL2DM has worked great on Linux for years; it would be nice to have "something" as to why it suddenly does not. I would guess that's all any of us really want here. It's obviously not a Linux issue, and Valve borked a working product without apparent resolve.

You're right, now we wait...

**Comment by MasterFrag** [comment]
Adding my voice to the list of people who can't launch the game now, with the binaries gone, all that's left is Proton, which VAC doesn't work with.

**Comment by angelbruni** [comment]
Also adding my voice here: can't play HL2:DM since the anniversary update.

**Comment by Bulletdust** [comment]
Also commenting that since the anniversary update I've been unable to play HL2DM under Linux.

**Comment by henk717** [comment]
Ok so what I am about to say is risky. This MAY BAN YOUR ACCOUNT. But for the brave souls who got an alt account with hl2dn during the sale or want to risk a VAC BAN there are 2 things that can be researched.

The first method would be running steam itself trough wine. This should work since its always traditionally worked because then the steam authentication stuff is also being emulated. This is the least likely of the two methods to get you vac banned but for me steam ubfer wine is super glitchy and ingame in the menu I get like 2fps. Someone would have to find a solution to that issue and then hope vac doesn't trigger on the wine binaries.

The second method is more likely to work but also a lot more dangerous. HL2DM got ported to the same engine as tf2. When they broke srcds we could get the servers running by borrowing tf2's files. You can likely use the files from TF2 to supplement any missing files. I don't know however if valve allows this at all with VAC and if not it will certainly see that as file tampering.

The only true solution is if valve adds the binaries back but if people test the above and don't get banned in a few weeks we'd at least have something.

**Comment by nixRidge** [comment]
> The first method would be running steam itself trough wine.

Just like old times, before Steam on Linux. I just gave it a shot, works fine enough, but I had to use wine-ge-8-26 provided by Lutris. And it's really not a great experience compared to what we had.

> You can likely use the files from TF2 to supplement any missing files.

Gave this a shot with -insecure but couldn't get it to boot. Might still be possible though, but I'm not risking it without the -insecure option, lol..

**Comment by Bulletdust** [comment]
What's really odd about this situation, is the fact that when the Anniversary Update dropped and I worked out on the same day that the Linux HL2DM client was failing to load, I switched to 'beta - Upcoming fixes' and the client loaded and ran perfectly. A day later 'beta - Upcoming fixes' was removed and no longer available, and when it did once again become available it was showing the exact same failure to launch issue as the 'stable' release. What are Valve doing?

**Comment by lflo5727** [comment]
Was excited to see the update potentially bringing an influx of players, but feels like most will be gone (if not already) by the time a fix is released for Linux.
When SteamOS is their own Linux-based OS its really surprising to see what looks like the update not even being tested on it, given it doesn't so much as launch.

**Comment by Ethorbit** [comment]
It's missing the hl2_linux64 binary to start the game and useless VAC prohibits Proton use in Multiplayer. How does Valve mess up this badly?

**Comment by headlesscyborg** [comment]
This is a situation that should never happen. I maintain a HL2DM server and when I get an information about a possible cheater or any othe issue I should look into, I have no way to access the game and can only reply that Valve broke the game. The server auto-updated to the 20th anniversary version and I'm locked out, the legacy branch is useless without downgrading the server. I'd expect an official statement from Valve within 24 hours maximum since this issue happened. All my PCs run either Arch or Ubuntu.

**Comment by Bulletdust** [comment]
This is getting ridiculous.

The beta did actually work when the Anniversary Update dropped, only for Valve to pull the binaries. No feedback has been provided by Valve as to just what on Earth is going on with HL2DM under Linux, as thus far there's been no fix - The binaries still aren't available.

Where's the feedback? Better still, where's the fix? Valve needs to do better.

**Comment by nixRidge** [comment]
I've tried to explain the issue to Steam support and after two of them told me the issue was because I wasn't using Ubuntu, one agent finally pulled through and told me to use the legacy branch, but when I explained why that's an issue, they told me the issue has been forwarded to the developers, and will update me when there's news.

That was a week ago though so we'll see where it goes.

**Comment by henk717** [comment]
Unfortunately forwarded to the developers tends to mean nothing will be done wiith it. The developers are already made aware of it trough multiple channels, I even mailed gabe. They could look here, they could look at the community forum, they could look at their inbox, they could have playtested the game or if a single dev in the building played the game they would have known.

Don't expect a fix anytime soon.

**Comment by scorpius2k1** [comment]
Simple workaround to get HL2DM working again under Linux:
https://github.com/scorpius2k1/hl2dm-anniversary-linux

I have tested on Arch, Ubuntu, and Debian, but it should work on other distributions as well as Snap/Flatpak. Essentially, all that is needed is the HL2 binary and a custom launch script. A few things, such as the intro video, MP3 audio, loading maps locally, and running local listen servers, do not work. So far, I have had no issues (or VAC bans) after using this workaround. However, as @henk717 mentioned previously, this is not officially supported by Valve, so you may want to use it with an alt account just to be on the safe side.

Hopefully this works well until Valve releases an official update for us Linux users 🐧

**Comment by 0xReprogrammable** [comment]
It's hilarious how valve didn't fix such a simple issue for 3 weeks, so the community had to :-D.

**Comment by nixRidge** [comment]
![bilde](https://github.com/user-attachments/assets/c6d61fb5-af90-49c4-87de-d40af8ff02cd)

HL2:DM got a private "frog" repo four days ago. If that's made by who I think it is, we might (don't hold your breath) be in for a fix sooner than later :)

**Comment by scorpius2k1** [comment]
> HL2:DM got a private "frog" repo four days ago. If that's made by who I think it is, we might (don't hold your breath) be in for a fix sooner than later :)

![image](https://github.com/user-attachments/assets/2c19b184-45cf-41f6-a805-65972c7ad764)
From what I understand, the repo is headed by misyltoad, who helped port TF2 to x64 and worked on the recent anniversary update. I've heard hearsay that the aptly named "frog" branch is upcoming updates porting HL2DM, DoD, and CS:S to x64 for Windows & Linux. The latest changelist seems to support that. Fingers crossed.

**Comment by nixRidge** [comment]
That's what I assume, and which is why I'm so optimistic :) She's awesome.

**Comment by Bulletdust** [comment]
This is awesome news! Thanks for the update.

**Comment by PcChip** [comment]
still broken :sob: 
I'm really curious how this happened in the first place

**Comment by gilesfun** [comment]
> still broken 😭 I'm really curious how this happened in the first place

Damn, over a month. I just installed Ubuntu 64-bit so hopefully steam support will actually look at my request this time.

**Comment by gilesfun** [comment]
A small update from steam support here - They are recommending basic troubleshooting so I assume it's not internally tracked as a known issue. 

![image](https://github.com/user-attachments/assets/b3af1385-8616-4b91-8a8b-500b6328b741)

I might get around to writing some stuff for them but if anyone more knowledgeable than me wants to contact them make sure you do it from Ubuntu or they will not want to talk to you:  

![image](https://github.com/user-attachments/assets/00635ecf-e054-4b3e-959d-a0ba991d2816)

**Comment by nixRidge** [comment]
@gilesfun It is most likely being worked on, but also very likely that it's mostly one person working on the fix if she is, so it'll take a while. Grain of salt and all that, but the private "frog" repository for HL2DM points to the same developer who worked on TF2's 64-bit update and tons more, especially for Linux stuff at Valve.

**Comment by gilesfun** [comment]
> @gilesfun It is most likely being worked on, but also very likely that it's mostly one person working on the fix if she is, so it'll take a while. Grain of salt and all that, but the private "frog" repository for HL2DM points to the same developer who worked on TF2's 64-bit update and tons more, especially for Linux stuff at Valve.

I have the same feeling as you... Just thought that if I can get steam support on the case Valve might fund extra people hours on the issue. (Though I actually have no idea how many people they have on call to fix Linux stuff in general, especially around this time of year, if ur reading this Valve hire more Linux people like all the linux people). 

Though there is also every chance that the Frog branch is coincidental to this issue, given the fix should(?) be a rather easy update according to the comments above, and that the Frog branch appear to modify the game, windows and linux depots. Though I do not doubt that when the Frog branch is ready this issue will be resolved simply for the fact that I trust that 'one person' to test on Linux *_before_* pushing :P

**Comment by mathew2214** [comment]
debian 12. also having this issue. switching to steam_legacy is the only way im able to play the game. 
Valve, please fix.

**Comment by Bulletdust** [comment]
Has anyone received any correspondence or updates on this matter? It feels like the native build of HL2DM has been abandoned for Linux.

**Comment by nixRidge** [comment]
No news. If it is being worked on, I speculate that it's most likely just that one developer working on it alone between her other projects.

**Comment by henk717** [comment]
Abandoned for now seems accurate. The frog branch was removed and no update has been done since 2 months. Its extremely easy to fix all valve has to do is put the binary in the client. Some people have gotten it to work by borrowing hl2_linux from singleplayer or other valve games. This may or may not get you vac banned.

**Comment by ihaveamac** [comment]
I don't know if the frog branch was actually removed, but private branches are no longer exposed through the API.

**Comment by henk717** [comment]
Ah thats good to know, then that potentially impending doom still is a worry for me. I really hope they don't break compatibility with the legacy branch further.

**Comment by Bulletdust** [comment]
> Abandoned for now seems accurate. The frog branch was removed and no update has been done since 2 months. Its extremely easy to fix all valve has to do is put the binary in the client. Some people have gotten it to work by borrowing hl2_linux from singleplayer or other valve games. This may or may not get you vac banned.

Anyone here care borrowed the hl2_linux from single player and want to report back as to whether VAC picked it up or not?

**Comment by b4n6-b4n6** [comment]
I can confirm [fix](https://github.com/scorpius2k1/hl2dm-anniversary-linux?tab=readme-ov-file#-support) works and I didn't get banned from online play

**Comment by Bulletdust** [comment]
> I can confirm [fix](https://github.com/scorpius2k1/hl2dm-anniversary-linux?tab=readme-ov-file#-support) works and I didn't get banned from online play

Yep, I've been using it and can confirm the fix works fine. Amazing that the community can fix the issue, but Valve themselves appear to be hopeless?

**Comment by Colin1224** [comment]
New Half-Life 2: Deathmatch update today, but the issue still isn't fixed.

**Comment by Mantas-2155X** [comment]
Can confirm it works with the latest update if you edit `hl2mp.sh` to remove the `exit 1` line inside the sniper runtime environment check and start the game via `./hl2mp.sh -game hl2mp -steam -novid` instead of the steam client.

EDIT: I was able to play both singleplayer and multiplayer servers

**Comment by Colin1224** [comment]
> Can confirm it works with the latest update if you edit `hl2mp.sh` to remove the `exit 1` line inside the sniper runtime environment check and start the game via `./hl2mp.sh -game hl2mp -steam -novid` instead of the steam client.


I can confirm this works, though it launches in native wayland mode which doesn't work great. So I used `SDL_VIDEO_DRIVER=x11` to force it back into using xwayland.

**Comment by NotHyper-474** [comment]
> Can confirm it works with the latest update if you edit `hl2mp.sh` to remove the `exit 1` line inside the sniper runtime environment check and start the game via `./hl2mp.sh -game hl2mp -steam -novid` instead of the steam client.
> 
> EDIT: I was able to play both singleplayer and multiplayer servers

Doesn't work for me on Arch Linux, it'll only print this
```
FATAL: It appears hl2mp was not launched within the Steam for Linux sniper runtime environment.
FATAL: Please consult documentation to ensure correct configuration, aborting.

WARNING: setlocale('en_US.UTF-8') failed, using locale:'C'. International characters may not work.
SDL video target is 'x11'
SDL video target is 'x11'
AppFramework : Unable to load module engine.so!
Unable to load interface VCvarQuery001 from engine.so
```

**Comment by PANCHO7532** [comment]
> Can confirm it works with the latest update if you edit `hl2mp.sh` to remove the `exit 1` line inside the sniper runtime environment check and start the game via `./hl2mp.sh -game hl2mp -steam -novid` instead of the steam client.
> 
> EDIT: I was able to play both singleplayer and multiplayer servers

For me it's still broken but with a different message
```
SDL video target is 'x11'
SDL video target is 'x11'
AppFramework : Unable to load module vguimatsurface.so!
Unable to load interface VGUI_Surface030 from vguimatsurface.so
```

**Comment by hyperoranje** [comment]
> For me it's still broken but with a different message
> 
> ```
> SDL video target is 'x11'
> SDL video target is 'x11'
> AppFramework : Unable to load module vguimatsurface.so!
> Unable to load interface VGUI_Surface030 from vguimatsurface.so
> ```

Forcing the game to run under Steam Linux Runtime Sniper fixed this issue for me.

To do this I opened ``~/.local/share/Steam/config/config.vdi`` (location might differ depending on Steam installation) with a text editor, searched for ``"320"`` (include the quotes in the search), looked for the occurrence with the properties ``name``, ``config``, and ``priority``, and changed the value of ``name`` to ``steamlinuxruntime_sniper``.
If you can't find the ``"320"`` entry with the ``name`` property, set a compatibility tool on HL2DM within steam and it will show up.

![Image](https://github.com/user-attachments/assets/a4bbd265-f66b-4691-af56-c2ac9625fe96)

After you've done that restart Steam and try launching the game.
The game launched and I was able to play online.
Doing this also makes the changes to ``hl2mp.sh`` unneeded, as it will recognize that Sniper is being used.

For reference I'm using Fedora 41 and the steam package from the rpmfusion-nonfree-updates repository, but I imagine this this works on other distributions and Steam installations.

**Comment by Colin1224** [comment]
> Forcing the game to run under Steam Linux Runtime Sniper fixed this issue for me.

Forcing sniper and nothing else worked for me on CachyOS.

I actually noticed that it wasn't running properly with sniper earlier, so I tried to set it as the compatibility tool like you can with scout, but the option wasn't there so I gave up. Good idea to just force it in the config.

**Comment by Snoucher** [comment]
Close the issue, the 64 bit update arrived. We can finally play hl2dm natively on linux.

**Comment by hyperoranje** [comment]
> the option wasn't there so I gave up. Good idea to just force it in the config.

Steam only displays Sniper as an option if the game explicitly supports it, for example Team Fortress 2.
Once Valve enables Sniper as an option for HL2DM and makes it the default compatibility tool for Linux, the issue can likely be closed.

**Comment by hyperoranje** [comment]
Seems like Valve just fixed it and Sniper is now the default runtime, workarounds should no longer be needed to run HL2DM on linux.
